drivers/i2c/Kconfig

```diff
@@ -128,4 +128,9 @@ config I2C_RV32M1_LPI2C
 	help
 	  Enable the RV32M1 LPI2C driver.
 
+config I2C_GET_CONFIG
+	bool "Enable get configuration API"
+	help
+	  This enables the API to get the current configuration of I2C.
```

Comment:
Please add more details, at least three lines. Here you could talk about what types of configuration can be obtained

Reply:
why do we need this kconfig? it is not common to guard drivers APIs with such kconfig, especially for something that is generic like this

Reply:
> 
> 
> why do we need this kconfig? it is not common to guard drivers APIs with such kconfig, especially for something that is generic like this

The question is whether we want to increase API size with calls that are not used often. Taking into account that I2C is stable and the function was not needed, so far, this may not be used too often.
Maybe we should consider opposite approach: provide Kconfig to remove API, in case when sb wants to cut out what is not used.

Reply:
I think the current approach is better because it doesn't force users to add additional configs to stay with the stable API. The same approach is in UART API, e.g. CONFIG_UART_LINE_CTRL or CONFIG_UART_DRV_CMD.